### PR TITLE
Feat: support scale of 9

### PR DESCRIPTION
**Pull request by sharafian**
Support a scale greater than that of XRP ledger. Amounts are simply rounded up in drops. Because the high-scale amounts are kept alongside claim signatures, though, the error is bounded at 1 drop. Uses the info at connection time to make sure the currency scale is matched with the client.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -535,18 +550,19 @@ class Plugin extends MiniAccountsPlugin {
...
     const keyPairSeed = util.hmac(this._secret, CHANNEL_KEYS + account.getAccount())
     const keyPair = nacl.sign.keyPair.fromSeed(keyPairSeed)
     const signature = nacl.sign.detached(encodedClaim, keyPair.secretKey)
 
-    debug(`signing outgoing claim for ${newBalance.toString()} drops on ` +
+    debug(`signing outgoing claim for ${newDropBalance.toString()} drops on ` +
       `channel ${clientChannel}`)
 
     const aboveThreshold = new BigNumber(util
       .xrpToDrops(account.getClientPaychan().amount))
```
Comment (dappelt) [inline index.js]: Not part of this PR, but shouldn't the client paychan's amount be reloaded after funding?

File: index.js [hunk 2]
```diff
@@ -46,11 +46,24 @@ class Plugin extends MiniAccountsPlugin {
     this._claimInterval = opts.claimInterval || util.DEFAULT_CLAIM_INTERVAL
     this._store = new StoreWrapper(opts._store)
     this._txSubmitter = createSubmitter(this._api, this._address, this._secret)
+    this._currencyScale = opts.currencyScale || 6
```
Comment (dappelt) [inline index.js]: You seem to assume that if `opts.currencyScale` is set, it is more than 6. Can you validate that the scale is higher than 6.

Alternatively, can you add tests that make sure everything works if the scale is < 6?

Comment (emschwartz) [reply index.js]: 👍 Also it should probably be `this._currencyScale = (typeof opts.currencyScale === 'number' ? opts.currencyScale : 6)` on the off chance that someone wants to use a scale of 0.

File: index.js [hunk 3]
```diff
@@ -46,11 +46,24 @@ class Plugin extends MiniAccountsPlugin {
...
+  xrpToBase (amount) {
+    return new BigNumber(amount)
+      .mul(Math.pow(10, this._currencyScale))
+      .toString()
+  }
+
+  baseToXrp (amount) {
+    return new BigNumber(amount)
+      .div(Math.pow(10, this._currencyScale))
+      .toFixed(6, BigNumber.ROUND_UP)
```
Comment (emschwartz) [inline index.js]: Wouldn't it be better to round down and just the extra as unsettled balance?

Comment (sharafian) [reply index.js]: The thing is, these plugins are emitting the amount of money they got in the base scale. If you round down it means the other side needs to round the amount they received down and then emit it. For some amounts this means a sendMoney call will send 0. Technically the error would still be bounded at one drop, but I think it's more confusing to work with if the sendMoney amount doesn't match the amount emitted by the other side

File: index.js (hunk 2 above)
Comment (sharafian) [reply index.js]: I almost added a check to make sure it's more than 6, but decided less than 6 should be allowed. I'll add a test for that case though.

File: index.js (hunk 1 above)
Comment (sharafian) [reply index.js]: can we put this as an issue?

File: index.js (hunk 2 above)
Comment (dappelt) [reply index.js]: Ok. Make sure to implement it the same way in `ilp-plugin-xrp-paychan` and `asym-client`.

File: index.js (hunk 1 above)
Comment (dappelt) [reply index.js]: Fixed in #30
